src/image_generation/image_generator.py

```diff
@@ -0,0 +1,76 @@
+# The class is a wrapper class for generating images from a prompt
+# The class uses the OpenAI API to generate images
+
+import openai
+
+class ImageGenerator:
+    MAX_PROMPT_LENGTH = 1000
+    MAX_IMAGE_SIZE = 1024
+
+    def get_user_prompt(self) -> str:
+        image_prompt = input('What do you want to generate an image of? ')
+        self.validate_prompt(image_prompt)
+        return image_prompt
+
+    def validate_prompt(self, image_prompt: str) -> None:
+        if not isinstance(image_prompt, str):
+            raise TypeError('Prompt must be a string')
+        
+        if len(image_prompt) == 0:
+            raise ValueError('Prompt must not be empty')
+        
+        if (len(image_prompt) > self.MAX_PROMPT_LENGTH):
+            raise ValueError('Prompt must be less than 1000 characters')
+
+    def validate_size(self, width: int, height: int) -> None:
```

Comment:
As this is a private method we should start the name with underscore. Function name should then become "_validate_size"